### PR TITLE
CN0531 Demo

### PR DESCRIPTION
**Pull request by adrimbarean**
CLI demo for the CN0531 project. The first two commits do not need review.

### CODE REVIEW

File: projects/ADuCM3029_demo_cn0531/src/ADuCM3029_demo_cn0531.c [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/*****************************************************************************
```
Comment (mchindri) [inline projects/ADuCM3029_demo_cn0531/src/ADuCM3029_demo_cn0531.c:1]: in commit message mein -> main

File: projects/ADuCM3029_demo_cn0531/src/cn0531.c [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+ * @param [in] dev	 	 	 - The device structure.
+ * @param [in] short_command - True to display the long command prompt,
+ *                             false to display the short command prompt.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+static int32_t cn0531_help_dac(struct cn0531_dev *dev, bool short_command)
+{
+	if (!short_command) {
+		return usr_uart_write_string(dev->cli_desc->uart_device,
+					     (uint8_t*)" dac_reg_read <addr>        - Read a DAC register.\n\r"
```
Comment (mchindri) [inline projects/ADuCM3029_demo_cn0531/src/cn0531.c:99]: This strings can be added as static global variables or defines. You will have more space for each line and the logic of the code will be easy to read.

File: projects/ADuCM3029_demo_cn0531/src/cn0531.c [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+ * @brief Application intialization.
+ * @param [out] device - Pointer to the device structure.
+ * @param [in] param - Pointer to the initialization structure.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t cn0531_init(struct cn0531_dev **device, struct cn0531_init_param *param)
+{
+	int32_t ret, val;
+	struct cn0531_dev *dev;
+	uint8_t app_name[] = {'C', 'N', '0', '5', '3', '1'};
```
Comment (mchindri) [inline projects/ADuCM3029_demo_cn0531/src/cn0531.c:347]: uint8_t app_name[6] = "CN0531";

Comment (mchindri) [reply projects/ADuCM3029_demo_cn0531/src/cn0531.c:347]: Just nitpick: I'am not sure about this :)

**Review by mchindri — APPROVED** [review]

**Comment by mchindri** [comment]
Nitpicl: Why not make instead of project and iio instead of cli?

**Comment by adrimbarean** [comment]
CLI project is still needed, but will make an IIO as well.

**Comment by adrimbarean** [comment]
V1:
 - fix commit message;
 - add expansive string literals as defines;
 - uint8_t app_name[6] = "CN0531"; seems to work.

**Pull request merged**